spec/xapian_db/erb_parser_spec.rb

```diff
@@ -0,0 +1,55 @@
+require File.expand_path(File.dirname(__FILE__) + '/../spec_helper')
+require 'yaml'
+require 'erb'
+require 'tempfile'
+
+describe '.adapter' do
+  let(:yaml_content_with_erb) do
+    <<-YAML
+      development:
+        adapter: <%= ENV.fetch('XAPIAN_DB_ADAPTER', 'generic') %>
+
+      test:
+        adapter: xapian
+    YAML
+  end
+
+  around do |example|
+    Tempfile.open('xapian_db.yml') do |file|
+      file.write(yaml_content_with_erb)
+      file.rewind
+      @config_file_path = file.path
+
+      example.run
+    end
+  end
+
+  context 'when loading adapter from a temporary YAML file with ERB' do
+    it 'loads the generic adapter based on environment variable' do
+      ENV['XAPIAN_DB_ADAPTER'] = 'generic'
+
+      db_config = if YAML.respond_to?(:unsafe_load_file)
```

Comment:
The test, does not test the code in lib/xapian_db/railtie.rb. It only tests the code here. The spec should really call real code.